extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -0,0 +1,173 @@
+    videoSettings['provider'] !== ALLOWED_AD_PROVIDER ||
+    !position ||
+    position === 'floating'
+  ) {
+    return;
+  }
+  const macros = getSrMacros(
+    media,
+    companionCampaignOptions['companionCampaignId'],
+    apesterElement,
+    consentObj
+  );
+  constructCompanionSrElement(
+    videoSettings['videoTag'],
+    position,
+    /** @type {!JsonObject} */ (macros),
+    apesterElement
+  );
+}
+
+/**
+ * @param {!JsonObject} video
+ * @return {?string}
+ */
+function getCompanionPosition(video) {
+  if (video) {
+    if (video['companion']['enabled']) {
+      return 'above';
+    }
+    if (video['companion_below']['enabled']) {
+      return 'below';
+    }
+    if (video['floating']['enabled']) {
+      return 'floating';
+    }
+  }
+}
+
+/**
+ * @param {string} videoTag
+ * @param {string} position
+ * @param {!JsonObject} macros
+ * @param {AmpElement} apesterElement
+ */
+function constructCompanionSrElement(
+  videoTag,
+  position,
+  macros,
+  apesterElement
+) {
+  const size = getCompanionVideoAdSize(apesterElement);
+  const ampBladeAd = createElementWithAttributes(
+    /** @type {!Document} */ (apesterElement.ownerDocument),
+    'amp-ad',
+    dict({
+      'width': size.width,
+      'height': '0',
+      'type': 'blade',
+      'layout': 'responsive',
+      'data-blade_player_type': 'bladex',
+      'servingDomain': 'ssr.streamrail.net',
+      'data-blade_macros': JSON.stringify(macros),
+      'data-blade_player_id': videoTag,
+      'data-blade_api_key': '5857d2ee263dc90002000001',
+    })
+  );
+
+  ampBladeAd.classList.add('amp-apester-companion');
+
+  const relativeElement =
+    position === 'below' ? apesterElement.nextSibling : apesterElement;
+  apesterElement.parentNode.insertBefore(ampBladeAd, relativeElement);
+
+  apesterElement.getResources().attemptChangeSize(ampBladeAd, size.height);
+}
+
+/**
+ * @param {AmpElement} apesterElement
+ * @return {{height: number, width: number}}
+ */
+export function getCompanionVideoAdSize(apesterElement) {
+  const adWidth = apesterElement./*REVIEW*/ clientWidth;
+  const adRatio = 0.6;
+  const adHeight = Math.ceil(adWidth * adRatio);
+  return {width: adWidth, height: adHeight};
+}
+
+/**
+ * @param {!JsonObject} interactionModel
+ * @param {?string} campaignId
+ * @param {AmpElement} apesterElement
+ * @param {!JsonObject} consentObj
+ * @return {{
+ * gdpr: ?number,
+ * page_url: string,
+ * param1: string,
+ * param2: ?string,
+ * param4: ?string,
+ * param6: ?string,
+ * param7: ?string,
+ * schain: ?string,
+ * user_consent: ?string
+ * }}
+ */
+function getSrMacros(interactionModel, campaignId, apesterElement, consentObj) {
+  const interactionId = interactionModel['interactionId'];
+  const publisherId = interactionModel['publisherId'];
+  const publisher = interactionModel['publisher'];
+
+  const pageUrl = Services.documentInfoForDoc(apesterElement).canonicalUrl;
+  const macros = {
+    param1: interactionId,
```

Comment:
Is the intent to prevent renaming on these? need to use the `dict()` function if so. It looks like the caller of this function just casts this as a `JsonObject`, but that isn't going to work.

Reply:
fixed.